File: solutions/226.invert-binary-tree/invert-binary-tree.js
// const invertTree = root => {
//   if (!root) return null

//   let tmp = invertTree(root.right)
//   root.right = invertTree(root.left)
//   root.left = tmp

//   return root
// }

const invertTree = root => {
  if (!root) return null
  ;[root.left, root.right] = [invertTree(root.right), invertTree(root.left)]
  return root
}

File: solutions/226.invert-binary-tree/invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if not root:
            return root

        right = root.right
        root.right = self.invertTree(root.left)
        root.left = self.invertTree(right)

        return root
